spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/DecisionTreeRegressor.java

```diff
@@ -0,0 +1,223 @@
+      }
+      Schema.Type features = inputField.getSchema().getType();
+      Preconditions.checkArgument(features.isSimpleType(), "Field to classify must be of simple type : String, int, " +
+        "double, float, long, bytes, boolean but was %s.", features);
+      Preconditions.checkArgument(!features.equals(Schema.Type.NULL), "Field to classify must not be of type null");
+    }
+    Schema.Field predictionField = inputSchema.getField(config.predictionField);
+    Preconditions.checkArgument(predictionField == null, "Prediction field must not already exist in the input " +
+      "schema.");
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      throw new IllegalArgumentException(String.format("Failed to find model to use for Regression. Location does " +
+                                                         "not exist: {}.", modelLocation));
+    }
+
+    final HashMap<String, Map<Object, Long>> categoricalFeaturesMap = new HashMap<>();
+    final String[] fields = config.features.split(",");
+
+    for (final String field : fields) {
+      Map<Object, Long> map = input.map(new Function<StructuredRecord, Object>() {
+        @Override
+        public Object call(StructuredRecord structuredRecord) throws Exception {
+          Schema schema = structuredRecord.getSchema().getField(field).getSchema();
+          Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+          if (!(type.equals(Schema.Type.INT) || type.equals(Schema.Type.DOUBLE) || type.equals(Schema.Type.FLOAT) ||
+            type.equals(Schema.Type.LONG))) {
+            return structuredRecord.get(field);
+          } else {
+            return null;
+          }
+        }
+      }).distinct().zipWithIndex().collectAsMap();
+
+      if (!(map == null ||
+        (map.size() == 1 && map.keySet().contains(null)))) {
+        categoricalFeaturesMap.put(field, map);
+      }
+    }
+
+    // load the model from a file in the model fileset
+    JavaSparkContext javaSparkContext = context.getSparkContext();
+    SparkContext sparkContext = JavaSparkContext.toSparkContext(javaSparkContext);
+    final DecisionTreeModel loadedModel = DecisionTreeModel.load(sparkContext, modelLocation.toURI().getPath());
+
+    JavaRDD<StructuredRecord> output = input.map(new Function<StructuredRecord, StructuredRecord>() {
+      @Override
+      public StructuredRecord call(StructuredRecord record) throws Exception {
+        List<Double> featureList = new ArrayList<Double>();
+        for (String field : fields) {
+          if (categoricalFeaturesMap.keySet().contains(field)) {
```

Comment:
does it make sense to store this instead of getting the keyset every loop? does the keyset change?